pandas/indexes/base.py

```diff
@@ -67,6 +67,8 @@ def _new_Index(cls, d):
     """ This is called upon unpickling, rather than the default which doesn't
     have arguments and breaks __new__
     """
+    if issubclass(cls, ABCPeriodIndex):
```

Comment:
why can't you just call ``__new__`` here like verything else? that's sort of the point, we are trying to eliminate special cases.